src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -316,6 +314,7 @@ public synchronized void close() throws IOException {
         // Close the underlying stream iff we own it...
         if (ownOutputStream) {
           out.close();
+          fs.close();
```

Comment:
If `fs` is null if we don't own the output stream, wouldn't it be safer to have a finally block and then call `fs.close()` when `fs != null`?

We also need to handle exceptions properly, as mentioned in an earlier comment.